### DISCUSSION TITLE
Autogen and front interface

### DISCUSSION THREAD

**Discussion by ggae2** [microsoft/autogen#4617] [Q&A] [answered]
I am beguinner on autogen and look for creating a simple chatbot to manage conversation with the human user, I create two agent assisstant and user proxy, I think the user proxy will paly the role of taking human input and assitant the agent that will answrer the human user. it works on terminal but my issue is I don't know how to display the conevsration on the interface as streamlit. I tried assisstant.get_replay but doens't work ,... any idea

**Comment by rysweet** [comment]
@peterychang didn't you do a streamlit demo?

@ggae2 I'm not super familiar with streamline but I tagged one of our team members that I think has some experience with it. You might also look at how autogen studio leverages user_proxy to see more of how it might be integrated with a UI

**Comment by SuMiaoALi** [comment]
I used the stable version of autogen 0.2.35 to build an Agents orchestration chat system for production environments.
Using UserProxyAgent as the entry proxy user, chat  with other agents, and capture chat_result and streaming to UI.
I will do some help if you need.

**Comment by cbsingh05iitkgp** [comment]
can someone share example for autogent 0.4 with streamlit.
I have a team of agents which runs aync..It takes input from user through user_proxy.
I am not sure, How do we need to manage user_proxy as user input and run stream for system message. If someone can share example , it will be helpful

**Reply by ekzhu** [reply]
While it is possible to use user proxy, would you consider instead using termination condition to stop the team, and take user input as the `task` parameter directly to the team in the next call to `run_stream`? 

Read: https://microsoft.github.io/autogen/stable/user-guide/agentchat-user-guide/tutorial/human-in-the-loop.html

For user proxy, you need to provide your own input function that take input from your UI. 

If you have worked out an example, would love to see a contribution in the form of a sample.

**Reply by sand-stratlytics** [reply]
I need help with this. I am facing a similar issue. I have doing a `SelectorGroupChat` with a planner and an executor agent. The planner agent, after generating the plan, needs human approval. I tried two different approaches. I first, added a `UserProxyAgent`, and though that I would be able to terminate whenever there is a `UserInputRequestedEvent`. Couldn't figure out how to add it as a terminating condition. I also tried without a user_agent but stuck. Could you please help me @ekzhu?

**Reply by ekzhu** [reply]
When the scenario is providing real-time approval, UserProxyAgent is suitable. I think you need to figure out how to pass in a custom input function that take input from StreamLit UI to the group chat.

On the other hand, instead of using UserProxyAgent, you can also use `SourceMatchTermination` so the group chat will terminate after the planner agent respond. This gives you a window to provide feedback as the next task to resume the group chat.

**Reply by cbsingh05iitkgp** [reply, marked as answer]
@sand-stratlytics You can see my example shared below. It is 2 agent interaction flow.
Whenever user intervention is required, Flow comes to Streamlit UI and then again team resume by taking input.
Summary:
Assitant handoff to "User". There is no actual user agent under team.
termination = HandoffTermination(target="user") | TextMentionTermination("TERMINATE")

https://github.com/cbsingh05iitkgp/AutoGenWithStreamLit/blob/main/SwarmwithAssitantv8Simple.py

**Comment by cbsingh05iitkgp** [comment]
Thanks @ekzhu 
I have written a sample code without using user_proxy which can help people to get started for streamlit and Autogen 0.4
https://github.com/cbsingh05iitkgp/AutoGenWithStreamLit

Sharing this example as i could not find any example. This is my first github upload as well.
